plasmapy/diagnostics/thomson.py

```diff
@@ -1,37 +1,268 @@
 """
-Defines the Thomson scattering analysis module as part of
-`plasmapy.diagnostics`.
+Defines the Thomson scattering analysis module as
+part of the diagnostics package.
 """
 
 __all__ = [
     "spectral_density",
+    "spectral_density_model",
 ]
 
 import astropy.constants as const
 import astropy.units as u
+import numbers
 import numpy as np
+import re
+import warnings
 
-from typing import List, Tuple, Union
+from lmfit import Model
+from typing import Any, Dict, List, Tuple, Union
 
 from plasmapy.formulary.dielectric import permittivity_1D_Maxwellian
 from plasmapy.formulary.frequencies import plasma_frequency
-from plasmapy.formulary.speeds import thermal_speed
-from plasmapy.particles import Particle
-from plasmapy.utils.decorators import validate_quantities
+from plasmapy.formulary.speeds import thermal_speed, thermal_speed_coefficients
+from plasmapy.particles import Particle, particle_mass
+from plasmapy.utils.decorators import (
+    bind_lite_func,
+    preserve_signature,
+    validate_quantities,
+)
+
+c_si_unitless = const.c.si.value
+e_si_unitless = const.e.si.value
+m_p_si_unitless = const.m_p.si.value
+m_e_si_unitless = const.m_e.si.value
+
 
 # TODO: interface for inputting a multi-species configuration could be
-# simplified using the plasmapy.classes.plasma_base class if that class
-# included ion and electron drift velocities and information about the ion
-# atomic species.
+#     simplified using the plasmapy.classes.plasma_base class if that class
+#     included ion and electron drift velocities and information about the ion
+#     atomic species.
+
+
+@preserve_signature
+def spectral_density_lite(
+    wavelengths,
+    probe_wavelength: numbers.Real,
+    n: numbers.Real,
+    Te: np.ndarray,
+    Ti: np.ndarray,
+    efract: np.ndarray = np.array([1.0]),
+    ifract: np.ndarray = np.array([1.0]),
+    ion_z=np.array([1]),
+    ion_mass=np.array([1]),
+    electron_vel=None,
+    ion_vel=None,
+    probe_vec=np.array([1, 0, 0]),
+    scatter_vec=np.array([0, 1, 0]),
+    instr_func_arr=None,
+):
```

Comment:
You're using mutable objects (i.e. `numpy` arrays) as default keywords here, which can lead to some [unexpected behavior](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments) and should generally be avoided.  This is pitfall almost everyone (including myself) runs into when learning Python.

You should replace all `numpy` array default values with `None` and use an if-clause to then set the default value to an array.

Reply:
Ah, thanks, yeah I keep forgetting this (probably because by luck I haven't been burned by it yet).